packages/react/src/dialog/root/useDialogRoot.ts

```diff
@@ -248,7 +250,7 @@ export namespace useDialogRoot {
     onOpenChange?: (
       open: boolean,
       event: Event | undefined,
-      reason: OpenChangeReason | undefined,
+      reason: DialogOpenChangeReason | undefined,
```

Comment:
We should use the publicly exported type here (`AlertDialog.Root.OpenChangeReason`). If `DialogOpenChangeReason` is used, devs won't know where to import it from.

Reply:
I suppose type only imports don't lead to circular dependency issues?

Reply:
The generator can't handle the public type of `Select.Root.OpenChangeReason`. To be fair, the `Actions` type is also not clear where it exists from at the moment, so I could just do `OpenChangeReason`

Reply:
Once I update the API extractor to handle types in namespaces, we shouldn't need to use `@type` annotations manually. Still, an ability to override could be useful sometimes.

Reply:
It does seem like they're still needed here

Reply:
This doesn't look right. I'm fixing the API extractor to handle type aliases better.